tidy3d/components/data/sim_data.py

```diff
@@ -145,7 +121,44 @@ def source_spectrum_fn(freqs):
             # remove user defined phase from normalization so its effect is present in the result
             return spectrum * np.conj(user_defined_phase)
 
-        return monitor_data.normalize(source_spectrum_fn)
+        return source_spectrum_fn
+
+    def renormalize(self, normalize_index: int) -> SimulationData:
+        """Return a copy of the :class:`.SimulationData` with a different source used for the
+        normalization."""
+
+        if normalize_index == self.simulation.normalize_index:
+            # already normalized to that index
+            return self.copy()
+
+        num_sources = len(self.simulation.sources)
+
+        if num_sources == 0:
+            # no sources present
+            normalize_index = None
+            log.warning("No sources present in simulation, not doing normalization.")
+
+        if normalize_index and (normalize_index < 0 or normalize_index >= num_sources):
+            # normalize index out of bounds for source list
+            raise DataError(
+                f"normalize_index {normalize_index} out of bounds for list of sources "
+                f"of length {num_sources}"
+            )
+
+        def source_spectrum_fn(freqs):
+            """Normalization function that also removes previous normalization if needed."""
+            new_spectrum_fn = self.source_spectrum(normalize_index)
```

Comment:
if `normalize_index` or `self.simulation.normalize_index` is `None`, does this still work? or do we need to set `new_spectrum_fn` / `old_spectrum_fn` to `lambda x: 1.0` here? I'm not understanding this edge case.

Reply:
It works because of this https://github.com/flexcompute/tidy3d/blob/b263febca19d6a3764dcfb7a36ebdffa7f7174c8/tidy3d/components/data/sim_data.py#L102-L103

Reply:
ah ok, I see